### PR TITLE
Add R.class files if they exist to provide better Android support

### PR DESCRIPTION
**Pull request by DonnieWest**

### CODE REVIEW

**Comment by wsdjeg** [comment]
you should not use classpathFiles += file
here `file` is the class file whole path ,not the classpath

**Comment by wsdjeg** [comment]
if you do so the classpath will be 
.:path/to/com/wsdjeg/Test.class:path/to/com/wsdjeg/Test2.class:path/to/com/wsdjeg/R.class

**Comment by wsdjeg** [comment]
it is a pity to close it,just need to change a litter,and i really have not know much about groovy,

File: classpath.gradle [hunk 1]
```diff
@@ -14,6 +14,16 @@ task classpath << {
       }
     }
 
+    def rjava = proj.getBuildDir().absolutePath + File.separator + "intermediates" + File.separator + "classes" + File.separator + "debug"
+    def rFiles = new File(rjava)
```
Comment (wsdjeg) [inline classpath.gradle]: i do not konw how to write in groovy,but in java it is 
```java
if (rFiles.existe()){
   let boolean flag = false;
   for(File f : rFiles){
      if f end with class let flag = true
   }
   if(flag){
  classpathFiles += rjava
 }
}
```

**Comment by DonnieWest** [comment]
I'll fix it up then next time I'm near my laptop

**Comment by DonnieWest** [comment]
@wsdjeg your insight made this work perfectly :)

@artur-shaik this should /finally/ be ready for merge

**Comment by artur-shaik** [comment]
@DonnieWest @wsdjeg thank guys. Can you write any docs that user should know to use it?

**Comment by DonnieWest** [comment]
What do you think needs to be added to the docs? This should work transparently with any gradle Android project

**Comment by wsdjeg** [comment]
do you try with any other proj which is not android?

**Comment by DonnieWest** [comment]
It should work with any gradle project that has any files that are similar to R.java files - lots of inner classes

**Comment by wsdjeg** [comment]
but the out put dir should has a properties to set it ,i has created a new gradle java proj, i found the out put dir different with android gradle proj

**Comment by wsdjeg** [comment]
just like maven's <outputdirectory>or <testoutputdirectory>

**Comment by wsdjeg** [comment]
@DonnieWest 
maybe you can try with my fork of syntastic
it provid better support for java checking of gradle proj
here is my fork
https://github.com/wsdjeg/syntastic

**Pull request merged**